packages/web-components/fast-foundation/src/flipper/flipper.template.ts

```diff
@@ -1,6 +1,27 @@
-import { ElementViewTemplate, html, when } from "@microsoft/fast-element";
+import { ElementViewTemplate, html } from "@microsoft/fast-element";
 import type { FASTFlipper, FlipperOptions } from "./flipper.js";
-import { FlipperDirection } from "./flipper.options.js";
+import type { FlipperDirection } from "./flipper.options.js";
+
+const templateCache = {};
```

Comment:
This is a great idea! Do we have a memoize option in templates or something in discussion @EisenbergEffect that we know of?

Reply:
This was @eisenbergeffect's idea - definitely curious on your follow-up as well. :)

Reply:
Directives like `when` and `repeat` already cache the templates internally, having the same effect. In your blog you achieve similar things by just storing templates as properties and then reusing them.

This particular case is special because the *only* dynamic part of the template is the `direction`, so a special optimization can be made to have two completely static templates swapped by one dynamic binding. I'm not sure we can generalize this but I do think this sort of pattern is worth an example in our documentation.

The principles behind this approach are:

* Avoid re-creating templates. That is relatively expensive.
* Reduce the number of bindings as much as possible.

Of course, it is more code, so that's a downside. But in this specific case, it's not that much more code, especially since we're using a simple function to construct the variants. In other situations with more or more complex templates, it might not be worth it.